repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala

```diff
@@ -20,22 +20,188 @@
+    *
+    * @param document the parsed saml response
+    * @return the issuer for the embedded assertions
+    * @throws SamlPolicyException if response is invalid
+    */
+  def validateResponseAndGetIssuer(document: Document): String = ???
+
+  /**
+    * Retrieves the policy from the configured endpoint. The caching will be handled elsewhere.
+    * There is a possiblity that this method will have to get split into two calls is we end up needing the raw policy for the os response mangling.
+    * I hope not, because that poops on what i'm trying to do with the cache at the moment.
+    *
+    * @param issuer the issuer
+    * @return the compiled xslt that represents the policy
+    * @throws SamlPolicyException for so many reasons
+    */
+  def getPolicy(issuer: String): XsltExecutable = ???
+
+  /**
+    * Applies the policy to the saml response.
+    *
+    * @param document the parsed saml response
+    * @param policy the xslt translation
+    * @return the translated document
+    * @throws SamlPolicyException if the translation fails
+    */
+  def translateResponse(document: Document, policy: XsltExecutable): Document = ???
+
+  /**
+    * Signs the saml response.
+    *
+    * @param document the translated saml response
+    * @return a signed saml response
+    * @throws SamlPolicyException if the signing fails
+    */
+  def signResponse(document: Document): Document = ???
+
+  /**
+    * Converts the saml response document into a servlet input stream for passing down the filter chain.
+    *
+    * @param document the final saml response
+    * @return the sinput stream that can be wrapped and passed along
+    */
+  def convertDocumentToStream(document: Document): ServletInputStream = ???
+
+  /**
+    * This is super vague because we don't know what it looks like yet.
+    *
+    * @return
+    */
+  def doResponseStuff() = ???
+
+  /**
+    * Evict from the cache as appropriate.
+    *
+    * @param atomEntry A {@link String} representation of an Atom entry. Note that Atom entries are XML elements,
+    */
+  override def onNewAtomEntry(atomEntry: String): Unit = ???
+
+  /**
+    * I suspect we don't care, but i could be wrong.
+    *
+    * @param event A value representing the new lifecycle stage of the system associated with the Feed that this
+    */
+  override def onLifecycleEvent(event: LifecycleEvents): Unit = ???
+
+  /**
+    * Stores the configuration and marks the filter as initialized.
+    * I'm going to have to initialize the cache and atom feed listener here, but it's not in the critical path for the moment.
+    *
+    * @param newConfiguration
+    */
+  override def configurationUpdated(newConfiguration: SamlPolicyConfig): Unit = super.configurationUpdated(newConfiguration)
+}
+
+case class SamlPolicyException(statusCode: Int, message: String, cause: Throwable) extends Exception(message, cause) {
+  def this(statusCode: Int, message: String) = this(statusCode, message, null)
```

Comment:
Rather than define another constructor, you may just want to provide a default for `cause` in the case class. That is:
`case class SamlPolicyException(statusCode: Int, message: String, cause: Throwable = null) extends Exception(message, cause)`

I think that still captures what you want to do, but it generates the appropriate `apply` method allowing users to write:
```
SamlPolicyException(200, "OK", new Exception())
SamlPolicyException(200, "OK")
```

Whereas the way it stands, a user must write:
```
SamlPolicyException(200, "OK", new Exception())
new SamlPolicyException(200, "OK")
```